### PR TITLE
doc: migrate documentation website to docusaurus

### PR DESCRIPTION
**Pull request by s-pace**
This PR:
- [x] Move .github folder and remove images
- [x] Remove old sources
- [x] Remove linting dependencies
- [X] Keep textlint
- [X] Keep LICENSE
- [X] Implement new form
- [X] Use docusaurus
- [ ] Make sure release is not impacted
- [ ] Make sure deployment will remain correct
- [x] Merge .gitignore
- [x] Import every documents .md from master

- [x] Removes Travis
- [x] Make sure to publish to netlify
- [ ] Make sure redirections are effective (.html, path, img)
- [ ] Update README
- [ ] The folder `docsearch` will be spread into the root/ folder of this repo (except .gitignore)
- [x] Revert c17fdb739eaf25496ba4f725af2dbb67fc8318d7 with updating docusaurus with the new release
- [x] Switch logo from black/light theme. Blocker https://github.com/facebook/docusaurus/issues/1924

Optionnally:
- [ ] Invest on using docusaurus plugin : @docusaurus/plugin-ideal-image

### CODE REVIEW

**Comment by endiliey** [comment]
what's the blocker for this ? 😉

**Comment by Shipow** [comment]
Hi @endiliey, everything run smooth with Docusaurus, super happy with the alpha so far!
We are holding the release a bit because the team is about to grow very soon. We need some organization for the communication and to integrate the CI of our next release.
I didn't check recently but last time we were just missing some meta for version and languages.
You can visit the new version WIP there: https://docsearch.netlify.com
More to come soon! Keep it up!

File: docsearch/sidebars.js [hunk 1]
```diff
@@ -7,9 +7,16 @@
...
   docs: {
-    'Getting Started': ['what-is-docsearch', 'how', 'apply'],
-    'Adding DocSearch to your UI': ['dropdown', 'styling', 'behavior'],
-    'Requirements, tips, FAQ': ['requirements','tips','faq','integrations'],
-    'Let\'s Get Technical': ['scraper','engine','config','css-helper','index','run-your-own']
-  },
+    "Getting Started": ["what-is-docsearch", "how", "who-can-apply", "apply"],
+    "Adding DocSearch to your UI": ["dropdown", "styling", "behavior"],
+    "Requirements, tips, FAQ": ["requirements", "tips", "faq", "integrations"],
+    "Let's Get Technical": [
```
Comment (Shipow) [inline docsearch/sidebars.js]: rename to "under the hood" ?

Comment (s-pace) [reply docsearch/sidebars.js]: Good idea. We used to have this label in the past, I will put it back

File: docsearch/plugins/my-loaders/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module.exports = function(context, options) {
+  return {
+    name: 'loaders',
+    configureWebpack(config, isServer) {
+      return {
+        mode:"development",
```
Comment (endiliey) [inline docsearch/plugins/my-loaders/index.js]: this will enforce webpack to be in dev mode even in production

Comment (s-pace) [reply docsearch/plugins/my-loaders/index.js]: Thanks @endiliey 
Resolved by removing it.

**Comment by s-pace** [comment]
Moved to https://github.com/algolia/docsearch-website

**Pull request closed without merging**